ecl/hql/hqlgram2.cpp

```diff
@@ -9766,9 +9766,22 @@ void HqlGram::checkDerivedCompatible(IIdAtom * name, IHqlExpression * scope, IHq
     }
 }
 
+bool HqlGram::checkSignedMacros()
+{
+    if (inSignedModule)
+        return true;
+
+    IFileContents * contents = lexObject->query_FileContents();
```

Comment:
Better to encapsulate the logic within the lexObject.  i.e.
```
lexObject->isImplicitilySigned()

bool HqlLex::isImplicitlySigned()
{
    return contents && contents->isImplicitlySigned();
}
```
However I am not sure that function will correctly handle all macros.  e.g.if an attributes calls an unsigned macro which uses a signed macro within it.

I would expect the following implementation:

```
bool HqlLex::isImplicitlySigned()
{
    if (contents && contents->isImplicitlySigned())
        return true;
    if (!inmacro)
        return false;
    return inmacro->isImplicitlySigned();
}
```